test/artifacts.jl

```diff
@@ -779,8 +779,11 @@ end
             Dict("0"^40 => ["not", "a", "string", "or", "dict"]),
             r"failed to parse entry",
         )
+
+        # reset DEPOT_PATH and force Pkg to reload what it knows about artifact overrides
         empty!(DEPOT_PATH)
         append!(DEPOT_PATH, old_depot_path)
+        Pkg.Artifacts.load_overrides(;force=true)
```

Comment:
Yikes, this is a nasty gotcha.  Although overrides are kind of not that widely used (and should generally be eschewed in favor of JLL preferences) it is still annoying that you need to make sure to clear this cache when the depot path changes.  Ideally, we would have a `set_depot_path!()` function that automatically does this kind of clearing for you.